tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -334,6 +370,50 @@ private function initializeData(array $initialData, array $expectedData = null)
         }
     }
 
+    private function prepareBulkWriteArguments($arguments)
+    {
+        $operations = [];
+        $operations['options'] = $arguments['options'];
+        foreach ($arguments['requests'] as $requests) {
+            if (isset($requests['arguments']['arrayFilters'])) {
+                if (isset($requests['arguments']['options'])) {
+                    array_push($requests['arguments']['options'], ['arrayFilters' => $requests['arguments']['arrayFilters']]);
```

Comment:
[`array_push()`](http://php.net/array_push) modifies its first argument by reference, but in this case, `$requests` is always a copy of the current element in `$arguments['requests']`, so your modification is never making it back to the `$arguments` value when you use it later in the function.

Two options:

 * Use a reference when iterating (e.g. `foreach ($values as &$value)`) so that any modifications to `$value` do modify the actual element in `$values`.
 * Track the array key while iterating (e.g. `foreach ($values as $i => $value)`) and refer to `$arguments['requests'][$i]` when you want to modify the current element within the loop.

This relates a bit to the zval behaviors discussed in the first half of [Memory Management](http://www.phpinternalsbook.com/zvals/memory_management.html) if you'd like to read more on the subject.